### PR TITLE
[ch307] Verify app can be configured to send via basic smtp servers

### PR DESCRIPTION
**Pull request by jnu**
Tested smtp configuration with a python `smtpd` server running locally.

The smtp connection worked fine, but to make configuration easier I fixed a limitation of pydantic's settings module to allow setting nested keys via the environment, and also mounted the `.env` file in the api container. Added some more documentation as well.

### CODE REVIEW

File: api/requirements.txt [hunk 1]
```diff
@@ -26,24 +27,28 @@ Jinja2==2.11.2
...
 python-dateutil==2.8.1
 python-dotenv==0.15.0
 python-multipart==0.0.5
+PyYAML==5.4.1
 requests==2.25.1
 six==1.15.0
 SQLAlchemy==1.3.23
 starlette==0.13.6
 toml==0.10.2
+tomlkit==0.5.11
```
Comment (nmanduj1) [inline api/requirements.txt:51]: I'm curious about these additional packages (PyYAML, tomlkit, more-itertools)- are you preemptively adding them for something or are they just additional debugging tools?

**Review by nmanduj1 — APPROVED** [review]
v nice! 🚀

File: api/requirements.txt (hunk 1 above)
Comment (jnu) [reply api/requirements.txt:51]: no these come `pydantic-settings`, which i used to fix the issue i described in the PR description:

> I fixed a limitation of pydantic's settings module to allow setting nested keys via the environment

also linked a GH issue below in the code.

Comment (nmanduj1) [reply api/requirements.txt:51]: gotcha- thanks!

**Pull request merged**